cmd/link.go

```diff
@@ -57,16 +58,34 @@ Please perform 'odo service create %s ...' before attempting to link the service
 			os.Exit(1)
 		}
 
-		// we also need to check whether there is a secret with the same name as the service
-		// the secret should have been created along with the secret
-		_, err = client.GetSecret(serviceName, projectName)
+		// if there is a ServiceBinding, then that means there is a secret (or perhaps there will be soon)
+		// which we can link to
+		_, err = client.GetServiceBinding(serviceName, projectName)
 		if err != nil {
-			fmt.Printf(`Secret %s should have been created along with the service
-If you previously created the service with 'odo service create', then you may have to wait a few seconds until OpenShift provisions it.
-If not, then please delete the service and recreate it using 'odo service create %s`, serviceName, serviceName)
+			fmt.Printf(`The service was not created via odo. 
+Please delete service and recreate it using 'odo service create %s'`, serviceName)
 			os.Exit(1)
 		}
 
+		if wait {
```

Comment:
I think this should be automatic rather than provisioned by user

Reply:
what if user don't provide wait

Reply:
That's a good point. WDYT @jorgemoralespou @kadel ?

Reply:
If the user doesn't supply `--wait` the command fails (which is the same as what we currently have)

Reply:
@geoand , considering noob user, I think it should be automatic (I mean no `--wait` flag)

Reply:
You have my vote :). Let's see what other think

Reply:
I don't know. I can see both options to make sense. I think that more important is to be consistent. If we make this wait by default that every other command where it makes sense should wait. That can be tricky, as I don't want to jump and start implementing it everywhere. Because of that, I would make this opt-it and we can add `--wait` only for some commands where we think it can be useful.

Reply:
Makes the most sense for the time being :)